CMakeLists.txt

```diff
@@ -436,8 +436,16 @@ configure_file(contrib/filter-lcov.py filter-lcov.py USE_SOURCE_PERMISSIONS COPY
 # Don't allow extended (non-ASCII) symbols in identifiers. This is easier for code review.
 try_append_cxx_flags("-fno-extended-identifiers" TARGET core_interface SKIP_LINK)
 
-try_append_cxx_flags("-ffile-prefix-map=A=B" TARGET core_interface SKIP_LINK
-  IF_CHECK_PASSED "-ffile-prefix-map=${PROJECT_SOURCE_DIR}/src=."
+# Avoiding the `-ffile-prefix-map` compiler option because it implies
+# `-fcoverage-prefix-map` on Clang or `-fprofile-prefix-map` on GCC,
+# which can cause issues with coverage builds, particularly when using
+# Clang in the OSS-Fuzz environment due to its use of other options
```

Comment:
I'm not sure this is a good inline comment, because it doesn't actually document the problem, just the side-effects.

Reply:
Mind suggesting a better wording for the comment?

Reply:
I don't really have one, given it's a somewhat obscure issue, but which is also why it'd be good to have proper documentation for. Anyone reading this in future isn't going to have any information other than, "don't use ffile-prefix-map otherwise it might break something in oss-fuzz". However given how long the build has been broken now, going to merge this to unbreak it.